### ISSUE TITLE
Function pointers don't specifcy calling convention

### ISSUE THREAD

**Issue by Susko3** [bottlenoselabs/SDL3-cs#493]
The [SDL definition](https://github.com/libsdl-org/SDL/blob/b90fed094f0936dcaddb9de26d527fb845372100/include/SDL3/SDL_events.h#L1071) notably includes `SDLCALL`, which is defined as `__cdecl`. 

```c
typedef int (SDLCALL *SDL_EventFilter)(void *userdata, SDL_Event *event);
```

So this should map to the following C# type (notice the `[Cdecl]`):
_ClangSharpPInvokeGenerator correctly generates this definition._

```cs
delegate* unmanaged[Cdecl]<void*, SDL_Event*, int>
```

Current code:
https://github.com/bottlenoselabs/SDL-cs/blob/158a06cccd26a36ac37dc001733d40e668d5c46a/src/cs/production/SDL/Generated/SDL.gen.cs#L3778-L3783

---

Tip: A cdecl callback function would be defined as:
```cs
[UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvCdecl) })]
private static int myFilter(void* userdata, SDL_Event* e)
{
    // do work
    return 1;
}
```

**Comment by lithiumtoast** [comment]
Yes, this is correct for .NET 5+. Before .NET 5+, it could not be done.  I have the type information of the calling convention but didn't update the C# code generator to use said information when adding function pointer support.

**Comment by lithiumtoast** [comment]
Thanks @Susko3,

This is fixed in the latest update as seen here https://github.com/bottlenoselabs/SDL3-cs/blob/34f93b31fa67a055c6ac3d2b2a6789a472a99dba/src/cs/production/Interop.SDL/Generated/SDL.g.cs#L10270
